### PR TITLE
STATE Macro harmonization

### PR DESCRIPTION
**Pull request by Soldia1138**
**When merged this pull request will:**
- Harmonize STATE Macros for radios
- Put STATE Macros to main component

### CODE REVIEW

File: addons/sys_prc117f/menus/fnc_changeMenu.sqf [hunk 1]
```diff
@@ -22,11 +22,11 @@ params["_newMenu"];
 
 
 // clear menu data
-SET_STATE("menuSelection" , 0);
+SET_STATE(menuSelection , 0);
```
Comment (Sniperhid) [inline addons/sys_prc117f/menus/fnc_changeMenu.sqf]: Due to the way the macros works this becomes some thing like `["menuSelection "`

File: addons/sys_prc152/menus/fnc_changeMenu.sqf [hunk 2]
```diff
@@ -22,15 +22,15 @@ params["_newMenu"];
 
 /*
 if(!([_newMenu] call FUNC(verifyIsMenu)) ) then {
-    _newMenu = GET_STATE_DEF("currentHome", GVAR(VULOSHOME));
+    _newMenu = GET_STATE_DEF(currentHome, GVAR(VULOSHOME));
 };
 */
 // clear menu data
-SET_STATE("menuSelection" , 0);
+SET_STATE(menuSelection , 0);
```
Comment (Sniperhid) [inline addons/sys_prc152/menus/fnc_changeMenu.sqf]: Same issue again -> `"menuSelection "`

**Review by Sniperhid — CHANGES_REQUESTED** [review]

File: addons/sys_prc152/menus/fnc_changeMenu.sqf (hunk 2 above)
Comment (Soldia1138) [reply addons/sys_prc152/menus/fnc_changeMenu.sqf]: Should be fixed by https://github.com/IDI-Systems/acre2/pull/48/commits/50c0de446a85dfa0d8d5fc99457ec22a2b752e47

**Review by Sniperhid — CHANGES_REQUESTED** [review]
You have fixed the issues however I discored a new issue with the 152/117F menus

```
private _saveName = MENU_SELECTION_VARIABLE(_menu);
SET_STATE(_saveName, _value);
```

is commonly used previously the value of `_saveName` would be used as the variable name e.g. `pgm_preset_number` now instead they are soted on `"_saveName"` instead

**Comment by Soldia1138** [comment]
According to my tests, this is working now.

**Review by Sniperhid — APPROVED** [review]

**Comment by Sniperhid** [comment]
Seconded my testing seems good as well! GJ JP

**Pull request merged**